.github/workflows/WE2E_tests.yaml

```diff
@@ -0,0 +1,123 @@
+name: Run WE2E Tests
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - rrfs_ci
+    types: 
+      - labeled
+
+# Use custom shell with -l so .bash_profile is sourced which loads intel/oneapi/setvars.sh, and lmod path 
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  
+  # The run_id is only available in this job if the workflow which builds the SRWA in build_SRWA.yml was successful.
+  get-run-id:
+    name: Get the GHA run_id of the SRWA build to pass to sequential jobs 
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    outputs: 
+      run-id: ${{ steps.get_run_id.outputs.run-id}}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: false
+      - id: get_run_id
+        run: |
+          FILE="${{ github.workspace }}/run_id.txt"
+          if [[ -f "$FILE"]]; then 
+            echo "::set-output name=run-id::$(cat ${{ github.workspace}}/run_id.txt)"
+          else 
+            echo "SRWA Build failed"
+            exit 1 
+          fi
+
+  set-matrix:
+    name: Set matrix of tests from machine_suites/ci.txt 
+    needs: get-run-id
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+ 
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: false
+      - id: set-matrix
+        run: | 
+          echo "::set-output name=matrix::$(cat ufs-srweather-app-${{ needs.get-run-id.outputs.run-id }}/regional_workflow/tests/WE2E/machine_suites/ci.txt | jq -R -s -c 'split("\n")[:-1]')"
+          echo 
+
+  create-experiment-directory:
+    needs: [ set-matrix ]
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    # We use THIS workflow's run id to set a new experiment directory, not the SRWA build workflow run_id 
+    steps:
+      - name: Create experiment directory 
+        run: |
+          mkdir -p /scratch1/ci-WE2E-test/${{ github.run_id }}
+
+  submit-all-tests:
+    name: Submit experiments to ./run_WE2E_tests.sh
+    needs: [ set-matrix, create-experiment-directory, get-run-id ]
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+
+    steps:
+      - name: Submit all experiments (WE2E tests) 
+        run: | 
+          cd ufs-srweather-app-${{ needs.get-run-id.outputs.run-id }}/regional_workflow/tests/WE2E/
+          module use -a /scratch1/miniconda3/contrib_miniconda3/modulefiles
+          module load miniconda3/4.11.0
+          conda activate regional_workflow
+          ./run_WE2E_tests.sh \
+            tests_file=machine_suites/ci.txt \
+            machine=aws \
+            account=account \
+            expt_basedir=/scratch1/ci-WE2E-test/${{ github.run_id }} \
+            exec_subdir=bin_intel/bin \
+            cron_relaunch_intvl_mnts=01 \
+            build_mod_fn=build_aws_intel
+
+  check-overall-we2e-status:
```

Comment:
@venitahagerty the script `./get_expts_status.sh` called below will check whether there was an error during workflow generation. If the argument provided as the `expts_basedir` does not exist, or is not a directory, the `./get_expts_status.sh`  script will print an error message an exit, and this GHA workflow will fail.